compiler/src/dotty/tools/dotc/core/TypeErasure.scala

```diff
@@ -118,7 +118,8 @@ object TypeErasure {
     semiEraseVCs <- List(false, true)
     isConstructor <- List(false, true)
     wildcardOK <- List(false, true)
-  } erasures(erasureIdx(isJava, semiEraseVCs, isConstructor, wildcardOK)) =
+  }
+  erasures(erasureIdx(isJava, semiEraseVCs, isConstructor, wildcardOK)) =
```

Comment:
same issue than with yield here, I would even say that the newline make the code easy to misinterpret.

Reply:
That just shows that `for` without `do` is evil. I would strongly suggest we get rid of it as fast as possible.

Reply:
That is one interpretation of the reality, but I fear you are biased in interpreting it that way. An equally valid interpretation of the reality is that the coding style enforced by the rewriting is actively harmful in some situations. A coding style that does not exhibit this issue is to systematically surround the body of a for loop with `{}` if the generators are themselves surrounded with `{}`. (This is for example part of the Scala.js coding style.)

Reply:
Even if we still need to discuss #7024 in general I believe we absolutely have to demand `do` after `while` and `for`. That part should be uncontroversial.